nifi-nar-bundles/nifi-slack-bundle/nifi-slack-services-rtm/src/main/java/org/apache/nifi/processors/slack/controllers/RTMSlackConnectionService.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.slack.controllers;
+
+import com.github.seratch.jslack.Slack;
+import com.github.seratch.jslack.api.rtm.RTMClient;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+
+import javax.websocket.DeploymentException;
+
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.lifecycle.OnDisabled;
+import org.apache.nifi.annotation.lifecycle.OnEnabled;
+import org.apache.nifi.annotation.lifecycle.OnShutdown;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.controller.AbstractControllerService;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.processor.Processor;
+import org.apache.nifi.processor.util.StandardValidators;
+@CapabilityDescription("Implementation of SlackConnectionService." +
+  "This service connects to the Real Time Messaging API of slack.")
+public class RTMSlackConnectionService extends AbstractControllerService implements SlackConnectionService {
+
+  private static final PropertyDescriptor API_TOKEN = new PropertyDescriptor.Builder()
+    .name("api-token")
+    .displayName("API token")
+    .description("Slack auth token required for Real Time Messaging API")
+    .sensitive(true)
+    .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+    .build();
```

Comment:
Please forgive my ignorance of Slack API, but let me confirm if my understandings are correct:
1. In order to use RTM, user needs to create a bot user?
2. I was able to user both `bot` and `user` types token with FetchSlack processor. However, while `bot` type api token works, I couldn't use `user` type token with this CS, I got following error. Only bot type token can be used here?
    ```
    java.lang.IllegalStateException: Couldn't fetch RTM API WebSocket endpoint. Ensure the apiToken value.
    ```
3. In order to receive messages in a channel, the bot app needs to be added to the channel. Is this correct?
4. What api token scopes are expected to be enabled?

I think it'd be super user friendly if we document these things.

Reply:
Based on the slack documentation https://api.slack.com/methods/rtm.connect bot and user token should work. I will check this issue. What is the convention on documenting external API-s? Copy the content, an extract of it or link it?

Reply:
I suggest adding 'additionalDetails.html' as it supports HTML docs so that we can link to the external API pages. Please refer UpdateAttribute for example for how to add additional details. https://nifi.apache.org/docs/nifi-docs/components/org.apache.nifi/nifi-update-attribute-nar/1.9.1/org.apache.nifi.processors.attributes.UpdateAttribute/additionalDetails.html